packages/cli/src/types.ts

```diff
@@ -53,12 +53,13 @@ export interface Configuration<Settings = unknown> {
   deploymentScriptDir?: string
   deploymentsDir?: string
   compilerOptions?: CompilerOptions
-  skipRecompile?: boolean
+  skipRecompileOnDeployment?: boolean
 
   networks: Record<NetworkId, Network<Settings>>
 
   enableDebugMode?: boolean
   forceRecompile?: boolean
+  skipRecompileIfDeployedOnMainnet?: boolean
 }
 
 export const DEFAULT_CONFIGURATION_VALUES = {
```

Comment:
I'm a bit hesitant about this because most of the contract compilation occurs during the development phase. Once the contract is deployed to the mainnet, the number of compilations should be much less.

If it's enabled by default, unnecessary checks could also be made on the mainnet during the development phase.

Reply:
Agreed. Let's not enable it by default.